srcjs/shinyapp.js

```diff
@@ -663,7 +663,11 @@ var ShinyApp = function() {
       // render the HTML and deps to a null target, so
       // the side-effect of rendering the deps, singletons,
       // and <head> still occur
-      exports.renderHtml($([]), message.content.html, message.content.deps);
+      try {
+        exports.renderHtml($([]), message.content.html, message.content.deps);
+      } catch(err) {
+        throw "The selector you chose could not be found in the DOM.";
```

Comment:
If I understand correctly, if `targets.length` is 0, it will always reach this `throw` statement. This is because in line 192 of output_binding_html.js, it calls `val.replace(self._reSingleton, findNewPayload)`, but `val` is `[]`, so there's no `replace` method.

If I'm right about that, then this code path (where `targets.length===0`) never worked, and that should be fixed. That should probably be done first. When would it be desirable to render this stuff when there are no targets?